File: client/src/types/JsonModels/Exercises/ExerciseLookupModel.ts
import type { ExerciseCategory } from "../Enums/ExerciseCategory";
import type { ExerciseDifficulty } from "../Enums/ExerciseDifficulty";
import type { ExerciseEquipment } from "../Enums/ExerciseEquipment";
import type { ExerciseLoadBasis } from "../Enums/ExerciseLoadBasis";
import type { ExerciseMovementPattern } from "../Enums/ExerciseMovementPattern";

export interface ExerciseLookupModel
{
	id: number;
	userId?: number;
	isGlobal: boolean;
	isPublic: boolean;
	name: string;
	slug: string;
	description?: string;
	imageUrl?: string;
	videoUrl?: string;
	primaryMuscleGroupId: number;
	primaryMuscleGroupName: string;
	secondaryMuscleGroupId?: number;
	secondaryMuscleGroupName?: string;
	equipment?: ExerciseEquipment;
	movementPattern?: ExerciseMovementPattern;
	difficulty?: ExerciseDifficulty;
	category?: ExerciseCategory;
	loadBasis?: ExerciseLoadBasis;
	aliases: string[];
	creatorUserId?: number;
	creatorDisplayName?: string;
	dateCreated: string;
}

File: server/FitMate.Core/JsonModels/Exercises/ExerciseLookupModel.cs
using FitMate.DB.Enums;

namespace FitMate.Core.JsonModels.Exercises;

public class ExerciseLookupModel
{
    public long Id { get; set; }
    public long? UserId { get; set; }
    public bool IsGlobal { get; set; }
    public bool IsPublic { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Slug { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string? ImageUrl { get; set; }
    public string? VideoUrl { get; set; }
    public long PrimaryMuscleGroupId { get; set; }
    public string PrimaryMuscleGroupName { get; set; } = string.Empty;
    public long? SecondaryMuscleGroupId { get; set; }
    public string? SecondaryMuscleGroupName { get; set; }
    public ExerciseEquipment? Equipment { get; set; }
    public ExerciseMovementPattern? MovementPattern { get; set; }
    public ExerciseDifficulty? Difficulty { get; set; }
    public ExerciseCategory? Category { get; set; }
    public List<string> Aliases { get; set; } = [];
    public long? CreatorUserId { get; set; }
    public string? CreatorDisplayName { get; set; }
    public DateTime DateCreated { get; set; }
}
